### PR TITLE
Fdg 7401 - Create GA4 events for DDP 

### PR DESCRIPTION
**Pull request by hmargy**
https://federal-spending-transparency.atlassian.net/browse/FDG-7401

no changes in test coverage

### CODE REVIEW

File: src/components/dataset-card/dataset-card.tsx [hunk 1]
```diff
@@ -39,17 +39,27 @@ const DatasetCard: FunctionComponent<DatasetCardProps> = ({
...
+        })
+      }
+      else {
         Analytics.event({
           category: `${context} Click`,
           action: `from ${referrer}`,
           value: dataset.name
-        })
+        });
+        // GA4 Data Layer - Related Dataset Click
```
Comment (wyattch) [inline src/components/dataset-card/dataset-card.tsx:56]: Right now, this triggers whenever a dataset card is clicked and not when it is just a related dataset

**Review by wyattch — CHANGES_REQUESTED** [review]
Currently, I am unable to get the cancel download tag to fire when clicking the button

File: src/components/dataset-card/dataset-card.tsx (hunk 1 above)
Comment (hmargy) [reply src/components/dataset-card/dataset-card.tsx:56]: good catch! I'll look into this

Comment (hmargy) [reply src/components/dataset-card/dataset-card.tsx]: changes have been added

**Review by wyattch — CHANGES_REQUESTED** [review]
Changes look great! One final thing I noticed is that when clicking on some datasets at random, I noticed that the date picker tag would fire when clicking on the dataset card from the search page. Could this be looked into?

**Comment by hmargy** [comment]
This is consistent with the current universal analytic logic

**Review by wyattch — APPROVED** [review]
Changes look good. Approved.

**Pull request merged**